external-crates/move/crates/move-compiler/tests/move_2024/parser/dot_incomplete.move

```diff
@@ -0,0 +1,18 @@
+module a::m {
+
+    public struct SomeStruct has drop {
+        some_field: u64
+    }
+
+    public struct AnotherStruct has drop {
+        another_field: SomeStruct
+    }
+
+    fun foo(s: AnotherStruct) {
+        let _tmp1 = s.;                // incomplete with `;` (next line should parse)
+        let _tmp2 = s.another_field.;  // incomplete with `;` (next line should parse)
+        let _tmp3 = s.another_field.   // incomplete without `;` (unexpected `let`)
+        let _tmp4 = s;
+        let _tmp = s.                  // incomplete without `;` (unexpected `}`)
```

Comment:
It' weird that we don't report this error.

Reply:
Good point. This was the original behavior, though, and I simply kept it. That being said, it definitely makes sense to be more precise here so I changed it to generate a diagnostic according to your suggestion.